### PR TITLE
Fix FGCs buffered dash bug

### LINKED ISSUE #830: Autoturn characters sometimes erroneously backdash after wavelanding

### PR DESCRIPTION
**Pull request by jobrien97**
Fixes an issue where the FGC characters would erroneously backdash if dash was buffered out of a waveland in certain situations.

Fixes #830

### CODE REVIEW

File: fighters/demon/src/status.rs [hunk 1]
```diff
@@ -302,6 +304,21 @@ unsafe extern "C" fn demon_attackcombo_main_loop_helper_second(fighter: &mut L2C
     status
 }
 
+// FIGHTER_STATUS_KIND_LANDING //
+
+#[status_script(agent = "demon", status = FIGHTER_STATUS_KIND_LANDING, condition = LUA_SCRIPT_STATUS_FUNC_INIT_STATUS)]
+pub unsafe fn landing_init(fighter: &mut L2CFighterCommon) -> L2CValue {
```
Comment (WuBoytH) [inline fighters/demon/src/status.rs]: Should have used Demon instead of Dolly here

File: fighters/ken/src/status.rs [hunk 2]
```diff
@@ -187,3 +191,18 @@ unsafe extern "C" fn ken_attack_main_loop(fighter: &mut L2CFighterCommon) -> L2C
...
+#[status_script(agent = "ken", status = FIGHTER_STATUS_KIND_LANDING, condition = LUA_SCRIPT_STATUS_FUNC_INIT_STATUS)]
+pub unsafe fn landing_init(fighter: &mut L2CFighterCommon) -> L2CValue {
+    app::FighterSpecializer_Dolly::update_opponent_lr_1on1(fighter.module_accessor, *FIGHTER_STATUS_KIND_TURN_DASH);
+    common::shoto_status::fgc_init_landing(fighter);
+    original!(fighter)
+}
+
+#[status_script(agent = "ken", status = FIGHTER_STATUS_KIND_LANDING, condition = LUA_SCRIPT_STATUS_FUNC_EXEC_STATUS)]
+pub unsafe fn landing_exec(fighter: &mut L2CFighterCommon) -> L2CValue {
+    common::shoto_status::fgc_exec_landing(fighter);
```
Comment (WuBoytH) [inline fighters/ken/src/status.rs]: Any reason why Terry and Kazuya have the FighterSpecializer call but Ryu and Ken don't?

Comment (jobrien97) [reply fighters/ken/src/status.rs]: added FighterSpecializer_Dolly call to Ryu/Ken (yes, it works)

**Comment by jobrien97** [comment]
I rewrote WAIT's main status, and later realized I didnt need it. So it's just there for reference

File: fighters/ken/src/status.rs (hunk 2 above)
Comment (WuBoytH) [reply fighters/ken/src/status.rs]: I figured it does, since Terry and Kazuya's unique 1 on 1 constant is the same value as Ryu's.

**Review by WuBoytH — APPROVED** [review]
lgtm

**Pull request merged**